modules/illuminBidAdapter.js

```diff
@@ -0,0 +1,336 @@
+import {_each, deepAccess, parseSizesInput, parseUrl, uniques, isFn} from '../src/utils.js';
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {BANNER, VIDEO} from '../src/mediaTypes.js';
+import {getStorageManager} from '../src/storageManager.js';
+import {config} from '../src/config.js';
+
+const DEFAULT_SUB_DOMAIN = 'exchange';
+const BIDDER_CODE = 'illumin';
+const BIDDER_VERSION = '1.0.0';
+const CURRENCY = 'USD';
+const TTL_SECONDS = 60 * 5;
+const UNIQUE_DEAL_ID_EXPIRY = 1000 * 60 * 15;
+const storage = getStorageManager({bidderCode: BIDDER_CODE});
+
+function getTopWindowQueryParams() {
+  try {
+    const parsedUrl = parseUrl(window.top.document.URL, {decodeSearchAsString: true});
+    return parsedUrl.search;
+  } catch (e) {
+    return '';
+  }
+}
+
+export function createDomain(subDomain = DEFAULT_SUB_DOMAIN) {
+  return `https://${subDomain}.illumin.com`;
+}
+
+export function extractCID(params) {
+  return params.cId || params.CID || params.cID || params.CId || params.cid || params.ciD || params.Cid || params.CiD;
```

Comment:
Question regarding these three extract methods. Is there a reason we are checking multiple combinations when the params are spec'd out to be `cId`, `pId` and `subDomain`? This seems unnecessary to me and throwing a warning during validation seems appropriate here if the property name is not correct

Reply:
@mmoschovas We are checking this for backward compatibility and minor typos of publishers.

Reply:
Maybe I am missing something. Is this not a new adapter? Im not sure where backwards compatibility is coming into play. I also understand the intent, I just do not understand why you are doing it considering it seems these are random alternative values and not comprehensive. Either way, I believe you should allow publishers who do not follow the specification to receive the error and not attempt to handle unknowns

Reply:
@mmoschovas 
I've changed the extract method to support only the given properties.
Can we merge this?